Validate PodSpec in EphemeralContainersUpdate

verb opened the pull request:
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Adding ephemeral containers to a pod only validates the list of ephemeral containers, but it's safer to validate the entire PodSpec because other parts of validation may have logic logic that depends on ephemeral containers.

#### Which issue(s) this PR fixes:
Fixes #105241

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

verb committed:
Validate PodSpec in EphemeralContainersUpdate

Previously this only validated the ephemeral containers, but it's safer
to validate the entire PodSpec in case other parts of validation add
logic that checks ephemeral containers.

verb commented:
/priority important-soon
/sig node
/test all

MadhavJivrajani commented:
/triage accepted

verb commented:
/assign @liggitt

liggitt commented on pkg/apis/core/validation/validation_test.go:
```diff
@@ -10616,9 +10616,31 @@ func TestValidatePodEphemeralContainersUpdate(t *testing.T) {
 	for _, test := range tests {
-		new := core.Pod{Spec: core.PodSpec{EphemeralContainers: test.new}}
-		old := core.Pod{Spec: core.PodSpec{EphemeralContainers: test.old}}
+		new, old := makePod(test.new), makePod(test.old)
```
It would be good to add a testcase that exercises a valid-in-isolation ephemeral container that is invalid in the context of a pod. One possibility could be a pod that specifies `securityContext.windowsOptions.hostProcess = true` in all other containers and tries to add an ephemeral container that does not set `securityContext.windowsOptions.hostProcess = true`

verb replied:
@liggitt  Good idea and thanks for suggesting a validation conflict: I didn't realize it was already possible. I've added this test case. PTAL.

verb committed:
Test ephemeral container/pod conflicting fields

This adds a test case to cover the scenario where the fields of an
ephemeral container conflict with other fields in the pod and must be
detected by full PodSpec validation.

liggitt commented:
/lgtm
/approve
